### PR TITLE
add configurable CE metadata extension prefix

### PR DESCRIPTION
**Pull request by cchayka-campsci**
## Proposed Changes

- allow the CloudEvent extension attributes "partition" and "offset" to have a customizable prefix

As brought up before in this issue https://github.com/knative-extensions/eventing-kafka-broker/issues/3992, a previous version of NodeJS `cloudevents` would throw an error for any extension > 20 chars in length.

We are in a tough position to update this package for a large number of legacy services, so we were hoping you could consider allowing this change.

Basically just adding a new env var "CE_METADATA_EXTENSION_PREFIX" to change the "knativekafka" part of the "knativekafkapartition" and "knativekafkaoffset" CE extensions.

**Release Note**

```release-note
:broom: allow customization of the prefix of the CE extensions
```

**Docs**

_previous pull request to the wrong branch:_ https://github.com/knative-extensions/eventing-kafka-broker/pull/4144

### CODE REVIEW

File: data-plane/dispatcher/src/main/java/dev/knative/eventing/kafka/broker/dispatcher/main/DispatcherEnv.java [hunk 1]
```diff
@@ -31,12 +32,17 @@ public class DispatcherEnv extends BaseEnv {
     public static final String EGRESSES_INITIAL_CAPACITY = "EGRESSES_INITIAL_CAPACITY";
     private final int egressesInitialCapacity;
 
+    public static final String CE_METADATA_EXTENSION_PREFIX = "CE_METADATA_EXTENSION_PREFIX";
```
Comment (matzew) [inline data-plane/dispatcher/src/main/java/dev/knative/eventing/kafka/broker/dispatcher/main/DispatcherEnv.java:35]: was there any feedback from the node.js library on why they fail - even the spec says `should` ? 

CC @Cali0707 

Asking here, instead of adding all this code for pluggable prefixes

Comment (cchayka-campsci) [reply data-plane/dispatcher/src/main/java/dev/knative/eventing/kafka/broker/dispatcher/main/DispatcherEnv.java:35]: The node.js library was updated successfully. We are in a tough position to update some of our legacy services so were hoping this change could be considered regardless.

Comment (Cali0707) [reply data-plane/dispatcher/src/main/java/dev/knative/eventing/kafka/broker/dispatcher/main/DispatcherEnv.java:35]: Yeah, we fixed it in the SDK since as you said @matzew  this was a "should" requirement. But, I guess that doesn't fix it if people are using a non-fixed version of the SDK

Comment (Cali0707) [reply data-plane/dispatcher/src/main/java/dev/knative/eventing/kafka/broker/dispatcher/main/DispatcherEnv.java:35]: Circling back here, I don't particularly mind having the extra code for the configuration if it improves interop with other systems. Any opinions @matzew @pierDipi ?
